In [1]:
import pandas as pd
import numpy as np
from scipy import stats

## **Question 1 --> Buyer Ratio**

## Problem Statement :
##### Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions

1 - Business Problem
Is the male-female buyer rations are similar across regions

2 - Data description
α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test

#### Chi-Square test

In [3]:
df = pd.read_csv("BuyerRatio.csv")
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

H0 == The male-female buyer rations are similar across regions       
H1 == The male-female buyer rations are not similar across regions

In [4]:
stats.chi2_contingency([df["East"], df["West"], df["North"], df["South"]])

Chi2ContingencyResult(statistic=1.5959455386610577, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  442.23468701],
       [ 146.81287862, 1518.18712138],
       [ 131.11756787, 1355.88243213],
       [  72.30424052,  747.69575948]]))

##### P value of Chi-Square test == 0.66 > α
###### Conclusion :
Thus HO is accepted.                                  
The male-female buyer rations are similar across regions

## **Question 2---->  Customer + order Form**

##### Problem Statement :         
TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

1 - Business Problem
Does the defective % varies significantly by centre ?

2 - Data description
α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test

##### Chi-Square test

In [5]:
df = pd.read_csv("Costomer+OrderForm.csv")

In [6]:
df

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

H0 == The defective % does not varies by centre        
H1 == The defective % does varies by centre

In [7]:
stats.chi2_contingency([df['Phillippines'].value_counts(), df['Indonesia'].value_counts(), df['Malta'].value_counts(), df['India'].value_counts()])

Chi2ContingencyResult(statistic=3.8589606858203545, pvalue=0.2771020991233144, dof=3, expected_freq=array([[271.75,  28.25],
       [271.75,  28.25],
       [271.75,  28.25],
       [271.75,  28.25]]))

P value of Chi-Square test == 0.277 > α

Thus HO is accepted.                        
The defective % does not varies significantly by centres

## **Question 3 ------> Cutlests**

**Problem Statement** :       
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions

1 - Business Problem
Is there significant difference in the diameter of the cutlet ?

2 - Data description
α == 0.05 (95% Confidence)

Y == Continious
X == Discrete

Is Y1 and Y2 normal ?

H0 = Y1 and Y2 are normal
H1 = Y1 and Y2 are not normal

#### **(1) Normality Test**

In [8]:
df = pd.read_csv("Cutlets.csv")
df

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [9]:
stats.shapiro(df["Unit A"])

ShapiroResult(statistic=0.9649459719657898, pvalue=0.31998491287231445)

P value for Unit A == 0.32 > α

In [10]:
stats.shapiro(df["Unit B"])

ShapiroResult(statistic=0.9727305769920349, pvalue=0.5225146412849426)

P value for Unit B == 0.52 > α

**HO is accepted. Thats is both Y1 and Y2 are normal**

#### **External Condition**
External condition are same.  
Thus we can perform Paired T Test

## **Model**

H0 == Mean for Y1 and Y2 are equal (There is no significance difference between diameter of the Culets)       
H1 == Mean for Y1 and Y2 are not equal (There is a significance difference between diameter of the Culets)

In [11]:
stats.ttest_rel(df["Unit A"], df["Unit B"])

TtestResult(statistic=0.7536787225614316, pvalue=0.4562300768038412, df=34)

P value of the Paired T Test is == 0.45 > α

**Thus H0 is accepted.
Mean of both Y1 and Y2 are equal**

## **Question 4 ---> LabTAT**

**Problem Statement :**  
A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch

Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level

1 - Business Problem
Is there a significant difference in the average Turn Around Time between laboratories ?

2 - Data description
α == 0.05 (95% Confidence)

Y == Continious
X == Discrete

Is Y1, Y2, Y3 and Y4 normal ?

H0 = Y1, Y2, Y3 and Y4 are normal
H1 = Y1, Y2, Y3 and Y4 are not normal



In [12]:
df = pd.read_csv("LabTAT.csv")

In [13]:
df

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [14]:
stats.shapiro(df["Laboratory 1"])

ShapiroResult(statistic=0.990183413028717, pvalue=0.5507769584655762)

In [15]:
stats.shapiro(df["Laboratory 2"])

ShapiroResult(statistic=0.9936320185661316, pvalue=0.8637352585792542)

In [16]:
stats.shapiro(df["Laboratory 3"])

ShapiroResult(statistic=0.988633930683136, pvalue=0.42045730352401733)

In [17]:
stats.shapiro(df["Laboratory 4"])

ShapiroResult(statistic=0.9913760423660278, pvalue=0.6619619131088257)

**HO is accepted. Thats is Y1, Y2, Y3, and Y4 are normal**

#### **Variance**
H0 == Variance of all 4 laboratories are the same    
H1 == Variance of all 4 laboratories are the not same

In [18]:
stats.levene(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

P Value of Variance test is == 0.051 > α

*H0 is accepted*

#### **Model**
H0 == Mean TAT for 4 laboratories equal (There is no significance difference between TAT of the laboratories)       
H1 == Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)

In [19]:
stats.stats.f_oneway(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])

<ipython-input-19-e40a46764b4e>:1: DeprecationWarning: Please use `f_oneway` from the `scipy.stats` namespace, the `scipy.stats.stats` namespace is deprecated.
  stats.stats.f_oneway(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])


F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

P value of the One way Anova test is == 2e-16 < α

Thus H1 is accepted.   
Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)